src/components/ChartJsWrapper/index.jsx

```diff
@@ -33,50 +34,70 @@ const ChartJsWrapper = ({
   type,
   chartHeight,
   spinnerSize,
-  missingDataError = false,
-}) => {
-  let showError;
+}) =>
+  (() => {
+    if (!data) {
+      return (
+        <div
+          style={{
+            lineHeight: spinnerSize,
+            textAlign: 'center',
+            width: spinnerSize,
+          }}>
+          <CircularProgress />
+        </div>
+      );
+    }
+
+    let error = null;
 
-  if (data && missingDataError) {
-    showError = data.datasets.every(dataset => {
-      const latestDataDate = new Date(dataset.data[dataset.data.length - 1].x);
+    data.datasets.forEach(dataset => {
+      const latestDataDate = new Date(
+        frum(dataset.data)
+          .sort('x')
+          .last().x
+      );
       const currentDate = new Date(); // get current date
       const timeDifference = Math.abs(
         currentDate.getTime() - latestDataDate.getTime()
       );
       const daysDifference = Math.ceil(timeDifference / (1000 * 3600 * 24));
 
-      return daysDifference > 3; // if days are more than 3 then show error
+      if (daysDifference > 3) {
+        error = new Error(
+          'This item has been missing data for at least 3 days.'
+        );
+      }
     });
-  }
 
-  return data ? (
-    <div className={classes.chartContainer}>
-      {showError && (
-        <CriticalErrorMessage
-          className={classes.errorPanel}
-          error="This item has been missing data for at least 3 days."
+    if (error) {
+      return (
+        <div className={classes.chartContainer}>
+          <ErrorMessage error={error}>
```

Comment:
Lines 77 to 83 are duplicated of 91 to 97.

You could merge the two by doing this (notice the line for errorMessage):
```javascript
    return (
      <div className={classes.chartContainer}>
        {title && <h2>{title}</h2>}
        <Chart
          type={type}
          data={data}
          height={chartHeight}
          options={generateOptions(options)}
        />
        {error && <ErrorMessage error={error}>}
      </div>
    );
```

You don't need to wrap the whole graph to make an error message obvious (see comparison):
<img width="786" alt="image" src="https://user-images.githubusercontent.com/44410/54609962-5e23d800-4a2a-11e9-89c7-3664bafeffe9.png">

Reply:
This is a legit concern, but what's happening here is a side effect of showing errors in general: **Now we get to see all the errors!**  There are two more problems, which we can solve in another set of PRs:

1.  This particular component is doing too much: this code redundancy will disappear once it is fixed: https://github.com/mozilla-frontend-infra/firefox-health-dashboard/issues/351
2. The type of error may dictate how best to reveal it to the user: this is partly solved with your earlier suggestion of `withErrorBoundary(specificErrorCompoenent)(wrappedCompoenent)`, which we can also do later.